src/redux/rockets/rockets.js

```diff
@@ -1,38 +1,48 @@
 // Actions
 const ROCKETS_FETCHED = 'space-travelers-hub/rockets/ROCKETS_FETCHED';
 const ROCKETS_RESERVED = 'space-travelers-hub/rockets/ROCKETS_RESERVED';
+const ROCKETS_CANCELLED = 'space-travelers-hub/rockets/ROCKETS_CANCELLED';
 
 // Make actions creators
 const fetchRocket = (payload) => ({
   type: ROCKETS_FETCHED,
   payload,
 });
 
-const reserveRocket = (status) => ({
+const reserveRocket = (id) => ({
   type: ROCKETS_RESERVED,
-  payload: status,
+  id,
+});
+
+const cancelRocket = (id) => ({
+  type: ROCKETS_CANCELLED,
+  id,
 });
 
 const initialState = [];
+let resState;
 
 // Make reducer
 const rocketsReducer = (state = initialState, action) => {
   switch (action.type) {
     case ROCKETS_FETCHED:
-      return [...state, action.payload];
-
+      return action.payload;
     case ROCKETS_RESERVED:
-      return [
-        ...state.map((mission) => {
-          if (mission.mission_id === action.mission.id) {
-            return { ...mission, reserved: action.mission.status };
-          }
-          return mission;
-        }),
-      ];
+      resState = state.map((item) => {
+        if (item.id !== action.id) return item;
+        return { ...item, reserved: true };
+      });
+      return resState;
+    case ROCKETS_CANCELLED:
+      resState = state.map((item) => {
+        if (item.id !== action.id) return item;
+        return { ...item, reserved: false };
+      });
+      return resState;
```

Comment:
You follow the instructions and apply an acceptable logic to make reservations active or inactive.💯